packages/core/src/llm/LLM.ts

```diff
@@ -674,21 +687,20 @@ export class Anthropic extends BaseLLM {
     };
   }
 
-  mapMessagesToPrompt(messages: ChatMessage[]) {
-    return (
-      messages.reduce((acc, message) => {
-        return (
-          acc +
-          `${
-            message.role === "system"
-              ? ""
-              : message.role === "assistant"
-                ? ANTHROPIC_AI_PROMPT + " "
-                : ANTHROPIC_HUMAN_PROMPT + " "
-          }${message.content.trim()}`
-        );
-      }, "") + ANTHROPIC_AI_PROMPT
-    );
+  getModelName = (model: string): string => {
+    if (Object.keys(AVAILABLE_ANTHROPIC_MODELS_WITHOUT_DATE).includes(model)) {
+      return AVAILABLE_ANTHROPIC_MODELS_WITHOUT_DATE[model];
+    }
+    return model;
+  };
+
+  formatMessages(messages: ChatMessage[]) {
+    return messages.map((message) => {
+      return {
+        content: message.content,
+        role: message.role as "user" | "assistant",
```

Comment:
This will throw error if there is non user or assistant type like, "system"

Reply:
Currently Anthropic only support user or assistant, they don't support system or others

Reply:
This is just a TS cast right?

But that's a good point that we should maybe check that all remaining messages are user or assistant, and not one of the other message types (and if they are throw an error)

But with the new changes, this function should currently be a noop.